src/Microsoft.Framework.PackageManager/Building/BuildManager.cs

```diff
@@ -192,17 +192,20 @@ public bool Build()
                 }
             }
 
-            if (!ScriptExecutor.Execute(project, "postbuild", GetScriptVariable))
+            if (success)
             {
-                WriteError(ScriptExecutor.ErrorMessage);
-                return false;
-            }
+                if (!ScriptExecutor.Execute(project, "postbuild", GetScriptVariable))
+                {
+                    WriteError(ScriptExecutor.ErrorMessage);
+                    success = false;
+                }
 
-            if (_buildOptions.GeneratePackages &&
-                !ScriptExecutor.Execute(project, "postpack", GetScriptVariable))
-            {
-                WriteError(ScriptExecutor.ErrorMessage);
-                return false;
+                if (_buildOptions.GeneratePackages &&
```

Comment:
At least, "postpack" runs only if package generation is done successfully. If "postpack" needs another condition, "postpack" should be solely responsible for that.